File: easy/3852_smallest_pair_with_different_frequencies/solution.cpp
#include <map>
#include <vector>

class Solution {
public:
  std::vector<int> minDistinctFreqPair(std::vector<int> &nums) {
    std::map<int, int> numsCounter;
    for (const int &num : nums)
      ++numsCounter[num];

    for (auto xIt = numsCounter.begin(); xIt != numsCounter.end(); ++xIt) {
      for (auto yIt = std::next(xIt); yIt != numsCounter.end(); ++yIt) {
        if (xIt->second != yIt->second)
          return {xIt->first, yIt->first};
      }
    }

    return {-1, -1};
  }
};

File: easy/3852_smallest_pair_with_different_frequencies/Solution.java
import java.util.Map;
import java.util.TreeMap;

class Solution {
  public int[] minDistinctFreqPair(int[] nums) {
    final Map<Integer, Integer> numsCounter = new TreeMap<>();
    for (final int num : nums) {
      numsCounter.put(num, numsCounter.getOrDefault(num, 0) + 1);
    }

    final Integer[] values = numsCounter.keySet().toArray(Integer[]::new);

    for (int i = 0; i < values.length; i++) {
      for (int j = i + 1; j < values.length; j++) {
        final int x = values[i];
        final int y = values[j];

        if (!numsCounter.get(x).equals(numsCounter.get(y))) {
          return new int[] {x, y};
        }
      }
    }

    return new int[] {-1, -1};
  }
}
